apps/minifront/src/components/swap/swap-form/token-swap-input.tsx

```diff
@@ -74,6 +75,15 @@ export const TokenSwapInput = () => {
     }
   };
 
+  const stakingTokenMetadata = useStakingTokenMetadata();
+
+  // Determine if the selected token is the staking token based on the current balances and metadata
+  const isStakingToken = hasStakingToken(
+    balancesResponses?.data,
+    stakingTokenMetadata.data,
+    assetIn,
+  );
```

Comment:
Looks like the selector can be re-used here too

Reply:
the above comment also applies here